v3/src/execution/__tests__/execution.test.ts

```diff
@@ -0,0 +1,366 @@
+import {
+  parseGraphqlRequest,
+  validateGraphqlRequest,
+  executeGraphqlRequest,
+  processGraphqlRequest
+} from "..";
+import { astSerializer } from "../../snapshotSerializers";
+import { VariableValues } from "apollo-server-core";
+import { makeExecutableSchema } from "graphql-tools";
+import { GraphQLError } from "graphql";
+
+expect.addSnapshotSerializer(astSerializer);
+
+// This is a handy trick for syntax highlighting GraphQL query strings in editors
+// like VS Code that depend on the gql`` tagged template literal.
+const gql = String.raw;
+
+// An array of tuples, each tuple representing a test.
+// A single tuple has the shape: [testName, { ...testParams }]
+type JestInput<T> = [
+  string,
+  T
+][];
+
+const validQueries: JestInput<{ query: string; operationName?: string; variables?: VariableValues }> = [
+  [
+    "basic - { people { name } }",
+    {
+      query: gql`
+      query Basic {
+        people {
+          name
+        }
+      }
+    `
+    }
```

Comment:
We could provide the expected result here in lieu of using snapshots. This could get unwieldy as we add more test cases, though I don't expect this is a behavior that will be changing often if ever. I personally enjoy the _convenience_ of snapshots, but I understand the concerns that people have with them and wouldn't mind giving up the convenience in favor of being deliberate and intentional about our expected test results.

Reply:
I like snapshots. I understand that they are impure, but we live in a fallen world, and in this fallen world, they make it easier to write tests, so I'm into them.